interface/src/app/tribe_client/tribe_client.js

```diff
@@ -10,32 +10,27 @@ angular.module('adage.tribe_client', [
 
   return {
     getUser: function() {
-      // only call this when the promise completes
+      // Only call this function after the promise completes
+      // to be able to get an actual user (instead of null).
       return user;
     },
+
     getPromise: function() {
       if (!promise) {
         promise = User.get({}, function(data) {
           if (data.meta.total_count !== 0) {
             user = data.objects[0];
           }
-        });
+        }).$promise;
       }
       return promise;
     },
+
     resetPromise: function() {
       // reset the promise in case we need to check the user again
       promise = null;
     },
-    setUser: function(newUser) {
```

Comment:
Exactly - yes, sorry, I should have mentioned this in the PR comments. This method was not being used anywhere and in fact I was thinking that it might not be the right way to go about implementing that functionality, even if we ever wanted it.